packages/Webkul/Product/src/Repositories/ProductRepository.php

```diff
@@ -505,15 +505,19 @@ public function getSuperAttributes($product)
      * @param  int  $categoryId
      * @return float
      */
-    public function getMaxPriceByCategory($categoryId)
+    public function getMaxPrice($params = [])
     {
         $customerGroup = $this->customerRepository->getCurrentGroup();
 
-        return $this->model
+        $query = $this->model
             ->leftJoin('product_price_indices', 'products.id', 'product_price_indices.product_id')
             ->leftJoin('product_categories', 'products.id', 'product_categories.product_id')
-            ->where('product_price_indices.customer_group_id', $customerGroup->id)
-            ->where('product_categories.category_id', $categoryId)
-            ->max('min_price');
+            ->where('product_price_indices.customer_group_id', $customerGroup->id);
+
+        if ($params['category_id'] != 'null') {
```

Comment:
Use `!empty()` or `isset()`. If the condition is still not matched then fix it from outside this method.

Reply:
sir fixed